python/cudf/cudf/core/column/column.py

```diff
@@ -470,13 +523,18 @@ def __setitem__(self, key, value):
         from cudf.core import column
 
         if isinstance(key, slice):
+            key_start, key_stop, key_stride = key.indices(len(self))
+            nelem = abs(key_stop - key_start)
```

Comment:
You need to assert that the stride is 1 or None before `nelem` is correct here.

Reply:
bad merge